[FLAKE]make test fails on osx and windows in travis CI

/kind bug
/kind flake

<!--

Welcome! - We kindly ask you to:

  1. Fill out the issue template below 
  2. Use the Google group if you have a question rather than a bug or feature request.

The group is at: https://groups.google.com/forum/#!forum/odo-users

Thanks for understanding, and for contributing to the project!

-->

## What versions of software are you using?

**Operating System:** 

**Output of `odo version`:** master

## How did you run odo exactly?

On OpenShift CI

## Actual behavior

Failed in getting the details of the changed file `TestWatchAndPush` on macOS. 

Also watch.go is creating some temporary files ```C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904\test_read_licenses.py216365071```  which is failed to get detected on windows.

## Expected behavior

`make test` should pass

## Any logs, error output, etc?

macOS travis CI
```
$ make test
go test -race github.com/openshift/odo/cmd/cli-doc github.com/openshift/odo/cmd/odo github.com/openshift/odo/pkg/application github.com/openshift/odo/pkg/application/labels github.com/openshift/odo/pkg/auth github.com/openshift/odo/pkg/catalog github.com/openshift/odo/pkg/component github.com/openshift/odo/pkg/component/labels github.com/openshift/odo/pkg/config 
[...]
ok  	github.com/openshift/odo/pkg/catalog	1.280s
E0131 09:18:12.583395    5459 watch.go:183] Failed getting details of the changed file /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/fabric8-analytics-license-analysis259913361/tests868419307/test_read_licenses.py612824246. Ignoring the change
E0131 09:18:12.585345    5459 watch.go:189] Ignoring event for file /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/fabric8-analytics-license-analysis259913361/tests868419307/test_read_licenses.py612824246 as details about the file couldn't be fetched
--- FAIL: TestWatchAndPush (11.10s)
    watch_test.go:690: Running test:  Case 1: Valid watch with list of files to be ignored with a append event
    watch_test.go:690: Running test:  Case 2: Valid watch with list of files to be ignored with a append and a delete event
    watch_test.go:690: Running test:  Case 3: Valid watch with list of files to be ignored with a create and a delete event
    watch_test.go:690: Running test:  Case 4: Valid watch with list of files to be ignored with a folder create event
    --- FAIL: TestWatchAndPush/Case_4:_Valid_watch_with_list_of_files_to_be_ignored_with_a_folder_create_event (1.03s)
 [...]
```
windows travis CI
```
$ make test
go test  -race github.com/openshift/odo/cmd/cli-doc github.com/openshift/odo/cmd/odo github.com/openshift/odo/pkg/application github.com/openshift/odo/pkg/application/labels github.com/openshift/odo/pkg/auth github.com/openshift/odo/pkg/catalog github.com/openshift/odo/pkg/component github.com/openshift/odo/pkg/component/labels github.com/openshift/odo/pkg/config github.com/openshift/odo/pkg/debug github.com/openshift/odo/pkg/devfile github.com/openshift/odo/pkg/devfile/adapters
[...]
E0303 10:03:25.148080    5044 watch.go:183] Failed getting details of the changed file C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904\test_read_licenses.py216365071. Ignoring the change
windows failed err CreateFile C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904\test_read_licenses.py216365071: The system cannot find the path specified.
E0303 10:03:25.152962    5044 watch.go:190] Ignoring event for file C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904\test_read_licenses.py216365071 as details about the file couldn't be fetched
windows ignore stat err CreateFile C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904\test_read_licenses.py216365071: The system cannot find the path specified.
E0303 10:03:25.153939    5044 watch.go:183] Failed getting details of the changed file C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904. Ignoring the change
windows failed err CreateFile C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904: The system cannot find the file specified.
E0303 10:03:25.153939    5044 watch.go:190] Ignoring event for file C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904 as details about the file couldn't be fetched
windows ignore stat err CreateFile C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904: The system cannot find the file specified.
E0303 10:03:25.153939    5044 watch.go:183] Failed getting details of the changed file C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904. Ignoring the change
windows failed err CreateFile C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904: The system cannot find the file specified.
E0303 10:03:25.153939    5044 watch.go:190] Ignoring event for file C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904 as details about the file couldn't be fetched
windows ignore stat err CreateFile C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\tests417416904: The system cannot find the file specified.
received [C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\__init__.py011991377 C:\Users\travis\AppData\Local\Temp\fabric8-analytics-license-analysis047095974\src445007821\read_licenses.py313912176] which is not same as expected list src/read_licenses.py,__init__.pyFAIL	github.com/openshift/odo/pkg/component	2.452s
ok  	github.com/openshift/odo/pkg/component/labels	1.047s
ok  	github.com/openshift/odo/pkg/config	1.147s
--- FAIL: Test_getDebugInfo (0.01s)
    --- FAIL: Test_getDebugInfo/case_3:_debug_port_not_listening (0.00s)
        info_test.go:315: getDebugInfo() got = {{OdoDebugInfo v1} 3844 testing-1 app nodejs-ex 5858 9001}, want {{ } 0    0 0}
        info_test.go:318: getDebugInfo() got1 = true, want false
    --- FAIL: Test_getDebugInfo/case_4:_the_process_is_not_running (0.00s)
        info_test.go:302: error while starting fake port listerner, cause: listen tcp 127.0.0.1:9001: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted.
FAIL
FAIL	github.com/openshift/odo/pkg/debug	0.377s
[...]
```

Issue ref : https://travis-ci.com/openshift/odo/jobs/281881842#L133 and https://travis-ci.com/openshift/odo/jobs/293340097#L110

--- a/pkg/component/watch_test.go
+++ b/pkg/component/watch_test.go
@@ -1,4 +1,4 @@
-// +build !windows,!osx
+// +build !osx
 
 package component
 
@@ -57,6 +57,7 @@ func setUpF8AnalyticsComponentSrc(componentName string, requiredFilePaths []test
 			// Preserve the new paths for further reference
 			fileProperties.FilePath = filepath.Base(realPath)
 			fileProperties.FileParent, _ = filepath.Rel(srcPath, filepath.Dir(realPath))
+			fileProperties.FileParent = filepath.FromSlash(fileProperties.FileParent)
 		}
 
 		// Perform mock operation as requested by the parameter
@@ -124,7 +125,7 @@ func mockPushLocal(client *occlient.Client, componentName string, applicationNam
 		// Verify every file in expected file changes to be actually observed to be changed
 		// If found exactly same or different, return from PushLocal and signal exit for watch so that the watch terminates gracefully
 		for _, gotChangedFile := range files {
-			wantedFileDetail := CompDirStructure[expChangedFile]
+			wantedFileDetail := CompDirStructure[filepath.FromSlash(expChangedFile)]
 			if filepath.Join(wantedFileDetail.FileParent, wantedFileDetail.FilePath) == gotChangedFile {
 				found = true
 			}
@@ -141,7 +142,7 @@ func mockPushLocal(client *occlient.Client, componentName string, applicationNam
 		// Verify every file in expected deleted file changes to be actually observed to be changed
 		// If found exactly same or different, return from PushLocal and signal exit for watch so that the watch terminates gracefully
 		for _, gotChangedFile := range delFiles {
-			wantedFileDetail := CompDirStructure[deletedFile]
+			wantedFileDetail := CompDirStructure[filepath.FromSlash(deletedFile)]
 			if filepath.Join(wantedFileDetail.FileParent, wantedFileDetail.FilePath) == filepath.Join(wantedFileDetail.FileParent, filepath.Base(gotChangedFile)) {
 				found = true
 			}

